### PR TITLE
Add `spotless` plugin

### PR DESCRIPTION
**Pull request by MaicolAntali**
### Purpose
Add the `spotless` plugins to enforce the Google Java Style Guide.

### Description
Related with #2284

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -124,6 +124,49 @@
...
+            <include>*.md</include>
+            <include>.gitignore</include>
+          </includes>
+          <trimTrailingWhitespace/>
+          <endWithNewline/>
+          <indent>
+            <spaces>true</spaces>
+            <spacesPerTab>4</spacesPerTab>
+          </indent>
+        </format>
```
Comment (MaicolAntali) [inline pom.xml:152]: I kept the `<format>` section, let me know if you prefer to remove it (or remove the `.md`)

File: pom.xml [hunk 2]
```diff
@@ -124,6 +124,49 @@
           </execution>
         </executions>
       </plugin>
+      <!-- Spotless plugin: keeps the code formatted following the google-java-styleguide -->
+      <plugin>
+      <groupId>com.diffplug.spotless</groupId>
+      <artifactId>spotless-maven-plugin</artifactId>
+      <version>2.40.0</version>
+      <executions>
```
Comment (Marcono1234) [inline pom.xml:132]: The indentation here for `<executions>` and `<configuration>` seems wrong; it should probably be indented by 2 additional spaces.

And `<formats>` should be indented by 4 additional spaces.

Comment (MaicolAntali) [reply pom.xml]: Oh yes, good catch!

**Comment by eamonnmcmanus** [comment]
Is this ready to merge?

**Comment by MaicolAntali** [comment]
If we want keep the `<format>` section, yes

**Review by eamonnmcmanus — APPROVED** [review]
OK. We can always adjust this later as needed.

Thanks again for doing this!

**Pull request merged**